autogen/extensions/agent_optimizer.py

```diff
@@ -0,0 +1,440 @@
+            else:
+                item = {
+                    "action_name": func.get("action_name"),
+                    "name": func.get("name"),
+                    "description": func.get("description"),
+                    "arguments": json.loads(func.get("arguments").strip('"')),
+                    "packages": func.get("packages"),
+                    "code": func.get("code"),
+                }
+            formated_actions.append(item)
+        actions = formated_actions
+
+        for action in actions:
+            name, description, arguments, packages, code, action_name = (
+                action.get("name"),
+                action.get("description"),
+                action.get("arguments"),
+                action.get("packages"),
+                action.get("code"),
+                action.get("action_name"),
+            )
+            if action_name == "remove_function":
+                incumbent_functions = [item for item in incumbent_functions if item["name"] != name]
+            else:
+                incumbent_functions = [item for item in incumbent_functions if item["name"] != name]
+                incumbent_functions.append(
+                    {
+                        "name": name,
+                        "description": description,
+                        "arguments": arguments,
+                        "packages": packages,
+                        "code": code,
+                    }
+                )
+
+        return incumbent_functions
+
+    def _construct_intermediate_prompt(self):
+        """
+        construct intermediate prompts.
+        """
+        if len(self._failure_functions_performance) != 0:
+            failure_experience_prompt = "We also provide more examples for different functions and their corresponding performance (0-100).\n The following function signatures are arranged in are arranged in ascending order based on their performance, where higher performance indicate better quality."
+            failure_experience_prompt += "\n"
+            for item in self._failure_functions_performance:
+                failure_experience_prompt += "Function: \n" + str(item["functions"]) + "\n"
+                failure_experience_prompt += "Performance: \n" + str(item["performance"]) + "\n"
+        else:
+            failure_experience_prompt = "\n"
+
+        if len(self._best_conversations_performance) != 0:
+            statistic_prompt = "The following table shows the statistical information for solving each task in each conversation and indicates, whether the result is satisfied by the users. 1 represents satisfied. 0 represents not satisfied."
+            statistic_prompt += "\n"
+            for item in self._best_conversations_performance:
+                statistic_prompt += str(item) + "\n"
+        else:
+            statistic_prompt = "\n"
+
+        return failure_experience_prompt, statistic_prompt
+
+    def _val_actions(self, actions, incumbent_functions):
```

Comment:
_validate_actions

no need to shorten the name here

Reply:
resolved